### ISSUE TITLE
Invalid handling of <br> tags

### ISSUE THREAD

**Issue by daladim** [mermaid-js/mermaid-cli#74]
**Describe the bug**
mermaid-cli generates invalid SVG when the mermaid file contains `<br/>` tags, although it is supported e.g. in the [Mermaid live editor](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbiAgc3ViZ3JhcGggc3ViXG4gICAgbm9kZShMaW5lIDE8YnIvPkxpbmUgMjxici8-TGluZSAzKVxuICBlbmRcbiAgIiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQiLCJ0aGVtZVZhcmlhYmxlcyI6eyJiYWNrZ3JvdW5kIjoid2hpdGUiLCJwcmltYXJ5Q29sb3IiOiIjRUNFQ0ZGIiwic2Vjb25kYXJ5Q29sb3IiOiIjZmZmZmRlIiwidGVydGlhcnlDb2xvciI6ImhzbCg4MCwgMTAwJSwgOTYuMjc0NTA5ODAzOSUpIiwicHJpbWFyeUJvcmRlckNvbG9yIjoiaHNsKDI0MCwgNjAlLCA4Ni4yNzQ1MDk4MDM5JSkiLCJzZWNvbmRhcnlCb3JkZXJDb2xvciI6ImhzbCg2MCwgNjAlLCA4My41Mjk0MTE3NjQ3JSkiLCJ0ZXJ0aWFyeUJvcmRlckNvbG9yIjoiaHNsKDgwLCA2MCUsIDg2LjI3NDUwOTgwMzklKSIsInByaW1hcnlUZXh0Q29sb3IiOiIjMTMxMzAwIiwic2Vjb25kYXJ5VGV4dENvbG9yIjoiIzAwMDAyMSIsInRlcnRpYXJ5VGV4dENvbG9yIjoicmdiKDkuNTAwMDAwMDAwMSwgOS41MDAwMDAwMDAxLCA5LjUwMDAwMDAwMDEpIiwibGluZUNvbG9yIjoiIzMzMzMzMyIsInRleHRDb2xvciI6IiMzMzMiLCJtYWluQmtnIjoiI0VDRUNGRiIsInNlY29uZEJrZyI6IiNmZmZmZGUiLCJib3JkZXIxIjoiIzkzNzBEQiIsImJvcmRlcjIiOiIjYWFhYTMzIiwiYXJyb3doZWFkQ29sb3IiOiIjMzMzMzMzIiwiZm9udEZhbWlseSI6IlwidHJlYnVjaGV0IG1zXCIsIHZlcmRhbmEsIGFyaWFsIiwiZm9udFNpemUiOiIxNnB4IiwibGFiZWxCYWNrZ3JvdW5kIjoiI2U4ZThlOCIsIm5vZGVCa2ciOiIjRUNFQ0ZGIiwibm9kZUJvcmRlciI6IiM5MzcwREIiLCJjbHVzdGVyQmtnIjoiI2ZmZmZkZSIsImNsdXN0ZXJCb3JkZXIiOiIjYWFhYTMzIiwiZGVmYXVsdExpbmtDb2xvciI6IiMzMzMzMzMiLCJ0aXRsZUNvbG9yIjoiIzMzMyIsImVkZ2VMYWJlbEJhY2tncm91bmQiOiIjZThlOGU4IiwiYWN0b3JCb3JkZXIiOiJoc2woMjU5LjYyNjE2ODIyNDMsIDU5Ljc3NjUzNjMxMjglLCA4Ny45MDE5NjA3ODQzJSkiLCJhY3RvckJrZyI6IiNFQ0VDRkYiLCJhY3RvclRleHRDb2xvciI6ImJsYWNrIiwiYWN0b3JMaW5lQ29sb3IiOiJncmV5Iiwic2lnbmFsQ29sb3IiOiIjMzMzIiwic2lnbmFsVGV4dENvbG9yIjoiIzMzMyIsImxhYmVsQm94QmtnQ29sb3IiOiIjRUNFQ0ZGIiwibGFiZWxCb3hCb3JkZXJDb2xvciI6ImhzbCgyNTkuNjI2MTY4MjI0MywgNTkuNzc2NTM2MzEyOCUsIDg3LjkwMTk2MDc4NDMlKSIsImxhYmVsVGV4dENvbG9yIjoiYmxhY2siLCJsb29wVGV4dENvbG9yIjoiYmxhY2siLCJub3RlQm9yZGVyQ29sb3IiOiIjYWFhYTMzIiwibm90ZUJrZ0NvbG9yIjoiI2ZmZjVhZCIsIm5vdGVUZXh0Q29sb3IiOiJibGFjayIsImFjdGl2YXRpb25Cb3JkZXJDb2xvciI6IiM2NjYiLCJhY3RpdmF0aW9uQmtnQ29sb3IiOiIjZjRmNGY0Iiwic2VxdWVuY2VOdW1iZXJDb2xvciI6IndoaXRlIiwic2VjdGlvbkJrZ0NvbG9yIjoicmdiYSgxMDIsIDEwMiwgMjU1LCAwLjQ5KSIsImFsdFNlY3Rpb25Ca2dDb2xvciI6IndoaXRlIiwic2VjdGlvbkJrZ0NvbG9yMiI6IiNmZmY0MDAiLCJ0YXNrQm9yZGVyQ29sb3IiOiIjNTM0ZmJjIiwidGFza0JrZ0NvbG9yIjoiIzhhOTBkZCIsInRhc2tUZXh0TGlnaHRDb2xvciI6IndoaXRlIiwidGFza1RleHRDb2xvciI6IndoaXRlIiwidGFza1RleHREYXJrQ29sb3IiOiJibGFjayIsInRhc2tUZXh0T3V0c2lkZUNvbG9yIjoiYmxhY2siLCJ0YXNrVGV4dENsaWNrYWJsZUNvbG9yIjoiIzAwMzE2MyIsImFjdGl2ZVRhc2tCb3JkZXJDb2xvciI6IiM1MzRmYmMiLCJhY3RpdmVUYXNrQmtnQ29sb3IiOiIjYmZjN2ZmIiwiZ3JpZENvbG9yIjoibGlnaHRncmV5IiwiZG9uZVRhc2tCa2dDb2xvciI6ImxpZ2h0Z3JleSIsImRvbmVUYXNrQm9yZGVyQ29sb3IiOiJncmV5IiwiY3JpdEJvcmRlckNvbG9yIjoiI2ZmODg4OCIsImNyaXRCa2dDb2xvciI6InJlZCIsInRvZGF5TGluZUNvbG9yIjoicmVkIiwibGFiZWxDb2xvciI6ImJsYWNrIiwiZXJyb3JCa2dDb2xvciI6IiM1NTIyMjIiLCJlcnJvclRleHRDb2xvciI6IiM1NTIyMjIiLCJjbGFzc1RleHQiOiIjMTMxMzAwIiwiZmlsbFR5cGUwIjoiI0VDRUNGRiIsImZpbGxUeXBlMSI6IiNmZmZmZGUiLCJmaWxsVHlwZTIiOiJoc2woMzA0LCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJmaWxsVHlwZTMiOiJoc2woMTI0LCAxMDAlLCA5My41Mjk0MTE3NjQ3JSkiLCJmaWxsVHlwZTQiOiJoc2woMTc2LCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJmaWxsVHlwZTUiOiJoc2woLTQsIDEwMCUsIDkzLjUyOTQxMTc2NDclKSIsImZpbGxUeXBlNiI6ImhzbCg4LCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJmaWxsVHlwZTciOiJoc2woMTg4LCAxMDAlLCA5My41Mjk0MTE3NjQ3JSkifX19)

**To Reproduce**
Save this as `basic.mermaid`:
```
graph TD
  subgraph sub
    node(Line 1<br/>Line 2<br/>Line 3)
  end
```

Grab the latest Docker image: `docker pull minlag/mermaid-cli`

Run mermaid-cli:
```
$ docker run -v /home:/mm minlag/mermaid-cli -i /mm/basic.mermaid -o /mm/basic.svg
yarn run v1.22.5
warning package.json: No license field
$ /app/node_modules/.bin/mmdc -p /puppeteer-config.json -i /mm/basic.mermaid -o /mm/basic.svg
Done in 0.42s.
```

**Expected behavior**
The SVG should be valid and identical to Mermaid's live editor output 

**Actual behaviour**
The SVG is invalid (no thumbnail preview in Nautilus), and Firefox displays the following error:
```
XML Parsing Error: mismatched tag. Expected: </br>.
Location: file:///home/basic.svg
Line Number 1, Column 3217:
<svg id="mermaid-1605261674894" width="100%" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" height="140" style="max-width: 150px;" viewBox="0 0 150 140"><style>#mermaid-1605261674894{font-family:"trebuchet ms",verdana,arial;font-size:16px;fill:#333;}#mermaid-1605261674894 .error-icon{fill:#552222;}#mermaid-1605261674894 .error-text{fill:#552222;stroke:#552222;}#mermaid-1605261674894 .edge-thickness-normal{stroke-width:2px;}#mermaid-1605261674894 .edge-thickness-thick{stroke-width:3.5px;}#mermaid-1605261674894 .edge-pattern-solid{stroke-dasharray:0;}#mermaid-1605261674894 .edge-pattern-dashed{stroke-dasharray:3;}#mermaid-1605261674894 .edge-pattern-dotted{stroke-dasharray:2;}#mermaid-1605261674894 .marker{fill:#333333;}#mermaid-1605261674894 .marker.cross{stroke:#333333;}#mermaid-1605261674894 svg{font-family:"trebuchet ms",verdana,arial;font-size:16px;}#mermaid-1605261674894 .label{font-family:"trebuchet ms",verdana,arial;color:#333;}#mermaid-1605261674894 .label text{fill:#333;}#mermaid-1605261674894 .node rect,#mermaid-1605261674894 .node circle,#mermaid-1605261674894 .node ellipse,#mermaid-1605261674894 .node polygon,#mermaid-1605261674894 .node path{fill:#ECECFF;stroke:#9370DB;stroke-width:1px;}#mermaid-1605261674894 .node .label{text-align:center;}#mermaid-1605261674894 .node.clickable{cursor:pointer;}#mermaid-1605261674894 .arrowheadPath{fill:#333333;}#mermaid-1605261674894 .edgePath .path{stroke:#333333;stroke-width:1.5px;}#mermaid-1605261674894 .flowchart-link{stroke:#333333;fill:none;}#mermaid-1605261674894 .edgeLabel{background-color:#e8e8e8;text-align:center;}#mermaid-1605261674894 .edgeLabel rect{opacity:0.5;background-color:#e8e8e8;fill:#e8e8e8;}#mermaid-1605261674894 .cluster rect{fill:#ffffde;stroke:#aaaa33;stroke-width:1px;}#mermaid-1605261674894 .cluster text{fill:#333;}#mermaid-1605261674894 div.mermaidTooltip{position:absolute;text-align:center;max-width:200px;padding:2px;font-family:"trebuchet ms",verdana,arial;font-size:12px;background:hsl(80,100%,96.2745098039%);border:1px solid #aaaa33;border-radius:2px;pointer-events:none;z-index:100;}#mermaid-1605261674894:root{--mermaid-font-family:"trebuchet ms",verdana,arial;}#mermaid-1605261674894 flowchart{fill:apa;}</style><g><g class="output"><g class="clusters"><g class="cluster" id="flowchart-sub-2" transform="translate(75,70)" style="opacity: 1;"><rect width="134" height="124" x="-67" y="-62"></rect><g class="label" transform="translate(0, -48)" id="mermaid-1605261674894Text"><g transform="translate(-13,-9)"><foreignObject width="26" height="18"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">sub</div></foreignObject></g></g></g></g><g class="edgePaths"></g><g class="edgeLabels"></g><g class="nodes"><g class="node default" id="flowchart-node-1" transform="translate(75,70)" style="opacity: 1;"><rect rx="5" ry="5" x="-32" y="-37" width="64" height="74" class="label-container"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-22,-27)"><foreignObject width="44" height="54"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">Line 1<br>Line 2<br>Line 3</div></foreignObject></g></g></g></g></g></g></svg>
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------^
```

Everything gets fixed if I remove the `<br/>` tags in the source mermaid file

**Comment by daladim** [comment]
Same issue if I use `<br>` tags instead of `<br/>` tags

**Comment by MindaugasLaganeckas** [comment]
This is issue is the same/related to https://github.com/mermaid-js/mermaid-live-editor/issues/26 and can be solved the same way.
@daladim : would you like to work on a PR?

**Comment by daladim** [comment]
Hi. Thanks for the info.
I am not sure I am experienced enough in JavaScript to make the PR myself, but let's see.

Jihchi has worked around this limitation [in mermaid.ink](https://github.com/jihchi/mermaid.ink/pull/9/files). It is very simple and effective. It indeed is a warkaround, but I can't see any use case where a `<br>` (that would not be an actual line break) would mistakenly be tampered by this workaround.

Otherwise, since `<br>`is correct HTML and `<br/>` is required for XML/XHTML compatibility, we might want to use an [XMLSerializer](https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer`) instead of a plain `.innerHTML`. This would be slightly more complex than the workaround above, but might work as well.

What is your opinion on this?

**Comment by MindaugasLaganeckas** [comment]
You are welcome to choose the solution yourself. A workaround is good enough, if it is faster to implement. We can always improve things :smile:

**Comment by MindaugasLaganeckas** [comment]
@daladim : thank you so much for your contributions so far. I have tried the svg file with the `<br>` and it does not give me any error in Firefox. https://github.com/mermaid-js/mermaid-cli/suites/1670560212/artifacts/31151542
The output of svg->png is different than just svg in Firefox for your test file. But no error is shown by the browser. 
My Christmas proposal; please, implement the fix, leave the test file, add only a single test (no for loop) and I will gladly accept your contribution. If needed, I will perform a manual verification of the output svg file.

**Comment by daladim** [comment]
It is strange that you don't get any error in Firefox on your side...
I have just checked the .zip file you mentioned. Both Firefox 83.0 on Ubuntu 20.04 and Firefox 84.0.1 on Windows 10 display the `Expected: </br>` error for me...
Anyway, I'll push the fix as you asked asap (and sorry for the Christmas delays!)

**Comment by MindaugasLaganeckas** [comment]
@daladim : can I close this issue?

**Comment by daladim** [comment]
Well, this depends on what you think about the associated MR #79 that you've merged. 

This merged MR should indeed fix the issue, but you mentioned errors that you kept having. As I can't remember how to test locally (it's been a long time, and I'm not a node/JS expert :-p ) I could not check that master was correct.

To be sure, I can test it if there was a tagged Docker image somewhere.

Please tell me how you want to proceed.
Thanks

**Comment by kyww** [comment]
Just run this command on the generated svg file as a workaround:
`sed 's#<br>#<br/>#g' -i output.svg`

**Comment by MindaugasLaganeckas** [comment]
Fixed by #171

**Comment by n0r** [comment]
I've just installed mermaid-cli via brew.
When running mermaid-cli against mermaid files containing `<br>` im still facing this error. @kyww's workaround fixes the previously rendered SVG.
